src/bio/bio_recovery.c

```diff
@@ -72,8 +74,23 @@ teardown_xs_bs(void *arg)
 	if (bxb->bxb_io_channel == NULL)
 		return;
 
-	/* Blobs (VOS pools) should have been close on faulty reaction */
-	D_ASSERT(d_list_empty(&bxb->bxb_io_ctxts));
+	/* When a normal device is unplugged, the opened blobs need be closed here */
+	d_list_for_each_entry(ioc, &bxb->bxb_io_ctxts, bic_link) {
+		if (ioc->bic_blob == NULL && ioc->bic_opening == 0)
+			continue;
+
+		opened_blobs++;
+		if (ioc->bic_closing || ioc->bic_opening)
+			continue;
+
+		bio_blob_close(ioc, true);
+	}
+ 
+	if (opened_blobs) {
```

Comment:
[Question]when we will free io_channel?

Reply:
When all opened blobs are closed, it could move on to putting io channel (see following code).